### PR TITLE
Robust symmetry and limb detection

### PR DESCRIPTION
**Pull request by Lars-DHondt-KUL**
### update
- new function to identify symmetry that is more robust
- improved detection of floating base translations, removed use of 'pelvis_ty' in favour of base_vertical
- legs no longer hardcoded to start at 'hip_r' and 'hip_l', this is now a setting (with default 'hip')
- idem for arms with 'acromial'

### how to test
- unit test for symmetry (w.r.t. old function) `./Tests/test_identify_kinematic_chain` 
- run simulation for 'Falisse_et_al_2022', 5 iterations should do. I had identical ipopt output w.r.t. example result.

### CODE REVIEW

File: PreProcessing/update_model_info.m
Comment (dgupta7) [inline PreProcessing/update_model_info.m:1]: Will the code work if we have pelvis torque actuators? These ones might not be a part of ActOpp.

File: Tests/test_identify_kinematic_chain.m [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+clear
+clc
+
+load('C:\GBW_MyPrograms\PredSim_test\Tests\Falisse_et_al_2022_Results\Falisse_et_al_2022_v1.mat')
+
+S = R.S;
+osim_path = model_info.osim_path;
+
+S.subject.base_joints_legs = 'hip';
+S.subject.base_joints_arms = {'acromial_r'};
```
Comment (dgupta7) [inline Tests/test_identify_kinematic_chain.m:11]: can we define {'hip_r','hip_l'} to S.subject.base_joints_legs. Need more description in readMe.

File: PreProcessing/identify_kinematic_chains.m [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+if strcmp(fl_b,'CustomJoint')
+    floating_base = CustomJoint.safeDownCast(floating_base);
+elseif strcmp(fl_b,'PlanarJoint')
+    floating_base = PlanarJoint.safeDownCast(floating_base);
+end
+
+sptr = floating_base.getSpatialTransform();
+
+for j=1:6
+
```
Comment (dgupta7) [inline PreProcessing/identify_kinematic_chains.m:191]: do we need j=1:6 if it is a planar joint as well?

**Review by dgupta7 — COMMENTED** [review]
Hi Lars, I have some questions and documentation requests. Please update those. Most of the code looks fine. Once you make the update, I will run it to further check. Thanks!

File: PreProcessing/update_model_info.m
Comment (Lars-DHondt-KUL) [reply PreProcessing/update_model_info.m:1]: Do we want the code to work with torque actuators between pelvis and ground? Torque actuators are meant to replace the net effect of muscles.
Updates in #119 will make it possible to add an external device that does this.

File: Tests/test_identify_kinematic_chain.m (hunk 1 above)
Comment (Lars-DHondt-KUL) [reply Tests/test_identify_kinematic_chain.m:11]: yes, I added some extra info to the settings in the readme

File: PreProcessing/identify_kinematic_chains.m (hunk 2 above)
Comment (Lars-DHondt-KUL) [reply PreProcessing/identify_kinematic_chains.m:191]: Planar joint for floating base only happens in 2D models, which are not supported by OpenSimAD anyways.
Support for 2D models is included in #107.

File: PreProcessing/update_model_info.m
Comment (dgupta7) [reply PreProcessing/update_model_info.m:1]: sorry, I meant trunk torque actuators.

**Review by dgupta7 — APPROVED** [review]
I looked through the code and it looks correct.
I ran the test_identify_kinematic_chain.m and it did not give any errors

File: PreProcessing/update_model_info.m
Comment (dgupta7) [reply PreProcessing/update_model_info.m:1]: i figured it out.

**Pull request merged**